be/src/vec/columns/column_json.h

```diff
@@ -0,0 +1,299 @@
+        return chars.allocated_bytes() + offsets.allocated_bytes();
+    }
+
+    void protect() override;
+
+    MutableColumnPtr clone_resized(size_t to_size) const override;
+
+    Field operator[](size_t n) const override {
+        assert(n < size());
+        return Field(&chars[offset_at(n)], size_at(n) - 1);
+    }
+
+    void get(size_t n, Field& res) const override {
+        assert(n < size());
+        res.assign_json(&chars[offset_at(n)], size_at(n) - 1);
+    }
+
+    StringRef get_data_at(size_t n) const override {
+        assert(n < size());
+        return StringRef(&chars[offset_at(n)], size_at(n) - 1);
+    }
+
+/// Suppress gcc 7.3.1 warning: '*((void*)&<anonymous> +8)' may be used uninitialized in this function
+#if !__clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
+
+    void insert(const Field& x) override {
+        const JsonField& s = doris::vectorized::get<const JsonField&>(x);
+
+        const size_t old_size = chars.size();
+        const size_t size_to_append = s.get_size() + 1;
+        const size_t new_size = old_size + size_to_append;
+
+        chars.resize(new_size);
+        memcpy(chars.data() + old_size, s.get_value(), size_to_append);
+        offsets.push_back(new_size);
+    }
+
+#if !__clang__
+#pragma GCC diagnostic pop
+#endif
+
+    void insert_from(const IColumn& src_, size_t n) override {
+        const ColumnJson& src = assert_cast<const ColumnJson&>(src_);
+        const size_t size_to_append =
+                src.offsets[n] - src.offsets[n - 1]; /// -1th index is Ok, see PaddedPODArray.
+
+        if (size_to_append == 1) {
+            /// shortcut for empty json
+            chars.push_back(0);
+            offsets.push_back(chars.size());
+        } else {
+            const size_t old_size = chars.size();
+            const size_t offset = src.offsets[n - 1];
+            const size_t new_size = old_size + size_to_append;
+
+            chars.resize(new_size);
+            memcpy_small_allow_read_write_overflow15(chars.data() + old_size, &src.chars[offset],
+                                                     size_to_append);
+            offsets.push_back(new_size);
+        }
+    }
+
+    void insert_data(const char* pos, size_t length) override {
+        const size_t old_size = chars.size();
+        const size_t new_size = old_size + length + 1;
+
+        chars.resize(new_size);
+        if (length) memcpy(chars.data() + old_size, pos, length);
+        chars[old_size + length] = 0;
+        offsets.push_back(new_size);
+    }
+
+    void insert_many_binary_data(char* data_array, uint32_t* len_array,
+                                 uint32_t* start_offset_array, size_t num) override {
+        for (size_t i = 0; i < num; i++) {
+            uint32_t len = len_array[i];
+            uint32_t start_offset = start_offset_array[i];
+            insert_data(data_array + start_offset, len);
```

Comment:
call insert_data for each row is right but poor performance, just follow ColumnString's batch way.